jobserver/views/job_requests.py

```diff
@@ -195,6 +195,14 @@ def get(self, request, *args, **kwargs):
         except (JobRequest.DoesNotExist, MultipleObjectsReturned):
             raise Http404
 
+        jobs = fetch(job_request.jobs.order_by("started_at"))
+
+        # we encode errors raised by job-runner when processing a JobRequest by
+        # misusing a Job, since that's our only method of returning messages
+        # from job-runner to job-server.  These misused jobs use the __error__
+        # action to differentiate themselves.
+        is_invalid = jobs.filter(action="__error__").exists()
```

Comment:
This line isn't really being tested at the moment, except as a side effect of fewer database queries being made. Is there an easy way to test that we're handling invalid jobs correctly? I know the tests are quite long at the moment..